### DISCUSSION TITLE
Coupling with FABM

### DISCUSSION THREAD

**Discussion by plazzari** [Parcels-code/Parcels#1730] [General]
Hello,
 my name is Paolo Lazzari (biogeochemical modeller).
 I am thinking about doing a test coupling between OceanParcels and [fabm](https://github.com/fabm-model/fabm/wiki)
 The first tests would be based on the outputs of gotm (1D vertical turbulence model).
 To avoid duplication of work, do you know if anyone is already working on similar topics?

Thank you very much!
 Paolo

**Comment by erikvansebille** [comment]
Hi @plazzari, thanks for considering to use Parcels and for reaching out with this question! No, as far as I know no-one has used Parcels before in combination with fabm. There is a bit of work on vertical diffusion in [this notebook](https://docs.oceanparcels.org/en/latest/examples/tutorial_diffusion.html), but nothing coupled to gotm. 

By the way: Is this work in the context of the [NECCTON project](https://neccton.eu), by any chance?

**Comment by plazzari** [comment]
Hi @erikvansebille,
 thank you for the useful example.
 I'm involved in the NECCTON project. Even if it wasn't originally planned within NECCTON, there could be some synergy with the recent activities of the new [nbfc](https://www.nbfc.it/en/project) centre to do some work on this task.
 From our side, we need to do some preliminary testing to understand the effort.
 We'll start with a fork and then try to add a vertical diffusion kernel, taking inspiration from the horizontal diffusion kernel you suggested.
Thanks very much!

**Reply by erikvansebille** [reply]
OK, thanks for the update. Note that @michaeldenes also works on the NECCTON project from our end, and has recently finished the https://plastic.oceanparcels.org/en/latest/ codebase; perhaps there's leverage there?

**Reply by plazzari** [reply]
Yes, I agree that the above link can be very useful, thank you!
